components/eam/src/chemistry/yaml/mo_adjrxt/adjrxt_input_ts_1400.yaml

```diff
@@ -0,0 +1,13 @@
+# Data at [lat=   39.0553303768561      , lon=   262.904774388088      , k=          54 , time step=        1400 ] 
+mam4xx:
+  function: adjrxt
+input:
+  fixed:
+    dt:    0.00
+    ncol: [         4]
+    inv: [0.2094011845E+020,0.1654269358E+020,0.4397424875E+019,0.4018888826E+017,0.8706654328E+006,0.5852783930E+008,0.8204468844E+008,0.1151649027E+013]
+    mtot: [0.2094011845E+020]
+    rate: [              NaN,0.4874449546E-011,0.1547236166E-011,0.9808320566E-012,0.3830369767E-011,0.1132613951E-010,0.1463821484E-011]
```

Comment:
does it work with NaN?

Reply:
It is a little worrying that NaNs are being passed as input. I know it's common practice in some circles to use NaNs as markers for "don't worry about this one lol", but I've always thought this is a horrible idea. Is there anything we can do to better understand what these NaNs mean, if anything?

Reply:
I can try to see where they are coming from....

Reply:
This variable is deliberately set to NaN [here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/d519f7a41af30314d654766ebe7e29a183b40d98/components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90#L389) to avoid any inadvertent use. Seems like the first index is never populated with a value and never used in the simulation. It seems like a signaling NaN, so our simulation should catch it if it is used anywhere since we always test it with debug flags turned on.  I think, in this case, we should add logic while writing the file, to identify and convert NANs to an invalid number. I would like to use a known invalid number so that we know that a NaN is being written out and the known number is just a mask. Let's choose -98765432 unless somebody has a better idea.